prac_05/emails.py

```diff
@@ -0,0 +1,147 @@
+"""
+CP1404/CP5632 Practical 05
+Email Register
+"""
+
+
+def main():
+    """This program allows you to store multiple emails, predict and confirm names upon entry."""
+    email_register = {}
+    email = " "
+    while email != "":
+        try:
+            email = input("Email: ")
+            if email != "":
+                validate_email(email)
+                predicted_name = extract_full_name(email)
+                accept_predicted = input("Is your name {}? (Y/n) ".format(predicted_name)).upper()
+                name = validate_name(predicted_name, accept_predicted)
+                # print("Stored name:", name)
+                store_in_email_register(email_register, name, email)
+                # print("Current register:", email_register)
+                # email = input("Email: ")
+                # validate_email(email)
+            else:
+                # print("You entered a blank email")
+                # print(email_register)
+                if email_register != "":
+                    display_email_register(email_register)
+        except ValueError:
+            print("The email register is currently empty")
+
+
+def validate_characters(email):
+    """Validate characters in the entered email"""
+    ending = email[-1]
+    if "@" not in email:
+        print("\tYour email must have the @ symbol")
+        return False
+    if "." not in email:
+        print("\tYour email must have the (.) symbol")
+        return False
+    if ending == ".":
+        print("\tYour email must not end with (.)")
+        return False
+    # print("Characters validated")
+    return True
+
+
+def count_dots(email):
+    """Count the dots in the entered email"""
+    dot_count = 0
+    for char in email:
+        if char == ".":
+            dot_count += 1
+    return dot_count
+
+
+def validate_corporate_formatting(email):
+    """Validate the entered email if it follows the format firstname.lastname@company.com"""
+    bypass = False
+    dot_count = count_dots(email)
+    valid_formatting = True
+    while not bypass and dot_count == 1 and valid_formatting:
+        print("Your email does not comply with the corporate standard formatting\n"
+              "\tExample: firstname.lastname@company.com\n")
+        proceed = input('Would you like to proceed with "{}"? (Y/n) '.format(email)).upper()
+        if proceed in ("Y", ""):
+            valid_formatting = True
```

Comment:
Instead of making a valid_formatting variable, could you not just make this a return True/False?

Reply:
I'm afraid It might break the while loop